uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -2540,5 +2540,29 @@ public function testCalculateAmountForDateForAContactUsingTheDefaultWorkPattern(
     $amount = LeaveBalanceChange::calculateAmountForDate($leaveRequest, new DateTime('2016-08-03'));
     $this->assertEquals(0, $amount);
   }
+
+  public function testCanDeleteTheBalanceChangesForALeavePeriodEntitlement() {
+    $leavePeriodEntitlement = $this->createLeavePeriodEntitlement();
+
+    LeaveBalanceChangeFabricator::fabricateForLeavePeriodEntitlement($leavePeriodEntitlement);
+    LeaveBalanceChangeFabricator::fabricateForLeavePeriodEntitlement($leavePeriodEntitlement);
+    LeaveBalanceChangeFabricator::fabricateForLeavePeriodEntitlement($leavePeriodEntitlement);
```

Comment:
Instead of creating a new fabricator method that it's unlikely to be used anywhere else other than here. I would probably be better to do this:

```php
$params = ['source_id' => $leavePeriodEntitlement->id, 'source_type' => LeaveBalanceChange::SOURCE_ENTITLEMENT];
LeaveBalanceChangeFabricator::fabricate($params);
LeaveBalanceChangeFabricator::fabricate($params);
LeaveBalanceChangeFabricator::fabricate($params);
```